File: src/brs/web/api/http/handler/SetAccountInfo.java
package brs.web.api.http.handler;

import brs.*;
import brs.services.ParameterService;
import brs.util.Convert;
import brs.web.api.http.common.APITransactionManager;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.JSONResponses.INCORRECT_ACCOUNT_DESCRIPTION_LENGTH;
import static brs.web.api.http.common.JSONResponses.INCORRECT_ACCOUNT_NAME_LENGTH;
import static brs.web.api.http.common.Parameters.DESCRIPTION_PARAMETER;
import static brs.web.api.http.common.Parameters.NAME_PARAMETER;

public final class SetAccountInfo extends CreateTransaction {

  private final ParameterService parameterService;
  private final Blockchain blockchain;

  public SetAccountInfo(ParameterService parameterService, Blockchain blockchain, APITransactionManager apiTransactionManager) {
    super(new LegacyDocTag[] {LegacyDocTag.ACCOUNTS, LegacyDocTag.CREATE_TRANSACTION}, apiTransactionManager, NAME_PARAMETER, DESCRIPTION_PARAMETER);
    this.parameterService = parameterService;
    this.blockchain = blockchain;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws SignumException {

    String name = Convert.nullToEmpty(req.getParameter(NAME_PARAMETER)).trim();
    String description = Convert.nullToEmpty(req.getParameter(DESCRIPTION_PARAMETER)).trim();

    if (name.length() > Constants.MAX_ACCOUNT_NAME_LENGTH) {
      return INCORRECT_ACCOUNT_NAME_LENGTH;
    }

    if (description.length() > Constants.MAX_ACCOUNT_DESCRIPTION_LENGTH) {
      return INCORRECT_ACCOUNT_DESCRIPTION_LENGTH;
    }

    Account account = parameterService.getSenderAccount(req);
    Attachment attachment = new Attachment.MessagingAccountInfo(name, description, blockchain.getHeight());
    return createTransaction(req, account, attachment);

  }

}

File: src/main/java/brs/http/SetAccountInfo.kt
package brs.http

import brs.Attachment
import brs.BurstException
import brs.Constants
import brs.DependencyProvider
import brs.http.JSONResponses.INCORRECT_ACCOUNT_DESCRIPTION_LENGTH
import brs.http.JSONResponses.INCORRECT_ACCOUNT_NAME_LENGTH
import brs.http.common.Parameters.DESCRIPTION_PARAMETER
import brs.http.common.Parameters.NAME_PARAMETER
import brs.util.Convert
import com.google.gson.JsonElement
import javax.servlet.http.HttpServletRequest

internal class SetAccountInfo(private val dp: DependencyProvider) : CreateTransaction(dp, arrayOf(APITag.ACCOUNTS, APITag.CREATE_TRANSACTION), NAME_PARAMETER, DESCRIPTION_PARAMETER) {

    @Throws(BurstException::class)
    internal override fun processRequest(request: HttpServletRequest): JsonElement {

        val name = Convert.nullToEmpty(request.getParameter(NAME_PARAMETER)).trim { it <= ' ' }
        val description = Convert.nullToEmpty(request.getParameter(DESCRIPTION_PARAMETER)).trim { it <= ' ' }

        if (name.length > Constants.MAX_ACCOUNT_NAME_LENGTH) {
            return INCORRECT_ACCOUNT_NAME_LENGTH
        }

        if (description.length > Constants.MAX_ACCOUNT_DESCRIPTION_LENGTH) {
            return INCORRECT_ACCOUNT_DESCRIPTION_LENGTH
        }

        val account = dp.parameterService.getSenderAccount(request)
        val attachment = Attachment.MessagingAccountInfo(dp, name, description, dp.blockchain.height)
        return createTransaction(request, account, attachment)
    }
}
